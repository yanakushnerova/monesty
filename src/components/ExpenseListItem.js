import React from 'react'
import { useHistory } from 'react-router-dom'

const ExpenseListItem = ({id, description, amount, createdAt}) => {
    const history = useHistory()
    
    return (
            <div>
                <p>{description}</p>
                <p>{amount} - {createdAt}</p>
                <button onClick={() => {
                     history.push(`/edit/${id}`)
                }}>edit expense</button>
            </div>
    )
}

export default ExpenseListItem
