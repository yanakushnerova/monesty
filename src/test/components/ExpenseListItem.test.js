import React from "react"
import ExpenseListItem from "../../components/ExpenseListItem"
import { shallow } from "enzyme"
import expenses from "../fixtures/expenses"

test('should render expense item', () => {
    const expense = {...expenses[1]}
    const wrapper = shallow(<ExpenseListItem expense={expense}/>)
    expect(wrapper).toMatchSnapshot()
})
