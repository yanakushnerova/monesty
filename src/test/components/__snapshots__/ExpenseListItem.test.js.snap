// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense item 1`] = `
<div>
  <p />
  <p>
     - 
  </p>
  <button
    onClick={[Function]}
  >
    edit expense
  </button>
</div>
`;
